src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -270,27 +286,42 @@ internal string FindBestVersionMatch(VersionRange versionRange, IEnumerable<stri
                 return matchingVersion?.ToString();
             }
 
-            // Check to see if there is a max bundle version set via hosting configuration, if yes then use that instead of the one
-            // available on VM or local machine. Only use MaximumBundleV3Version or MaximumBundleV4Version if the version configured
-            // by the customer resolved to version higher than the version set via hosting config.
-            if (!string.IsNullOrEmpty(configOption.MaximumBundleV3Version)
-                && matchingVersion?.Major == ScriptConstants.ExtensionBundleV3MajorVersion)
+            return matchingVersion?.ToString();
+        }
+
+        // Limit the bundle version range so that it is higher than 4.2.0 and lower than 4.22.0
+        private VersionRange GetCappedVersion(int dotnetVersion, VersionRange versionRange, string bundleId)
+        {
+            if (dotnetVersion == ScriptConstants.DotNetVersionSix
+                    && bundleId == ScriptConstants.DefaultExtensionBundleId
+                    && versionRange.MinVersion.Major == ScriptConstants.ExtensionBundleV4MajorVersion)
             {
-                var maximumBundleV3Version = NuGetVersion.Parse(configOption.MaximumBundleV3Version);
-                matchingVersion = matchingVersion > maximumBundleV3Version ? maximumBundleV3Version : matchingVersion;
-                return matchingVersion?.ToString();
+                NuGetVersion cappedMinversion = new(ScriptConstants.CappedMinimumBundleV4VersionExclusive);
+                NuGetVersion cappedMaxVersion = new(ScriptConstants.CappedMaximumBundleV4VersionExclusive);
+
+                var minversion = versionRange.MinVersion < cappedMinversion ? cappedMinversion : versionRange.MinVersion;
+                var maxversion = versionRange.MaxVersion > cappedMaxVersion ? cappedMaxVersion : versionRange.MaxVersion;
+
+                // Use the original inclusion values if it is within the min or max cap range.
+                bool isMinInclusive = versionRange.MinVersion > cappedMinversion && versionRange.IsMinInclusive;
+                bool isMaxInclusive = versionRange.MaxVersion < cappedMaxVersion && versionRange.IsMaxInclusive;
+
+                versionRange = new VersionRange(minversion, isMinInclusive, maxversion, isMaxInclusive);
             }
+            return versionRange;
+        }
 
-            if (!string.IsNullOrEmpty(configOption.MaximumBundleV4Version)
-                && matchingVersion?.Major == ScriptConstants.ExtensionBundleV4MajorVersion)
+        private NuGetVersion GetMaxCappedBundleVersionFromHostingConfig(string bundleId, int bundleVersion, int dotnetVersion, FunctionsHostingConfigOptions configOption)
```

Comment:
use the `TryGetXXX` (e.g., `TryGetEnvironmentBundleVersion` pattern here and avoid returning `null` from this method)